lib/compiler-singlepass/src/config.rs

```diff
@@ -19,7 +19,7 @@ impl Singlepass {
     /// specified.
     pub fn new() -> Self {
         Self {
-            enable_nan_canonicalization: true,
+            enable_nan_canonicalization: false,
```

Comment:
I think since singlepass is mainly directed for blockchains, which need deterministic results by default, this value should be true by default.